ext/extconf.rb

```diff
@@ -7,8 +7,55 @@
   $CPPFLAGS += $CXXFLAGS
 end
 
+brew = find_executable('brew')
+
+# Automatically use homebrew to discover thrift package if it is
+# available and not disabled via --disable-homebrew.
+use_homebrew = enable_config('homebrew', brew)
+
+if use_homebrew
+  homebrew_package = with_config('homebrew-thrift-package', 'thrift@0.9')
+  $stderr.puts "using Homebrew to locate thrift package '#{homebrew_package}'"
+
+  thrift_prefix = `#{brew} --prefix #{homebrew_package}`.strip
+
+  unless File.exists?(thrift_prefix)
+    $stderr.puts "#{thrift_prefix} does not exist"
+    $stderr.puts "To resolve, `brew install #{homebrew_package}` or pass"
+    $stderr.puts '--with-homebrew-thrift-package=<package> to this build to specify'
+    $stderr.puts 'an alternative thrift package to build against. e.g.:'
+    $stderr.puts
+    $stderr.puts '  $ bundle config --global build.sparsam --with-homebrew-thrift-package=mythrift'
+
+    raise "Homebrew package #{homebrew_package} not installed"
+  end
+
+  $stderr.puts "using Homebrew thrift at #{thrift_prefix}"
+  $CFLAGS << " -I#{thrift_prefix}/include"
```

Comment:
as an FYI, I have a separate PR out that I haven't gotten over the finish line that switches from directly modifying `$CFLAGS` and instead uses mkmf's helper method `append_cflags` when using a new enough version of mkmf -- [see this comment](https://github.com/airbnb/sparsam/pull/28/files#r765189467). Dunno if you also want to make these changes here, or I can follow-up with them later!

Reply:
👍🏻 sounds good, I'll rebase my branch on yours once we merge this